tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/PublicAssetSetMethodContract.scala

```diff
@@ -1163,4 +1164,238 @@ trait PublicAssetSetMethodContract {
+           |        "destroy": ["$invalidPublicAssetId"]
+           |      }, "0"
+           |    ]
+           |  ]
+           |}""".stripMargin)
+    .when()
+      .post
+    .`then`
+      .statusCode(HttpStatus.SC_OK)
+      .contentType(JSON)
+      .extract
+      .body()
+      .asString()
+
+    // Then the request should return notDestroyed
+    assertThatJson(response)
+      .inPath("methodResponses[0]")
+      .isEqualTo(
+        s"""[
+           |  "PublicAsset/set",
+           |  {
+           |    "accountId": "$ACCOUNT_ID",
+           |    "oldState": "$${json-unit.ignore}",
+           |    "newState": "$${json-unit.ignore}",
+           |    "notDestroyed": {
+           |      "$invalidPublicAssetId": {
+           |        "type": "invalidArguments",
+           |        "description": "Invalid UUID string: $invalidPublicAssetId"
+           |      }
+           |    }
+           |  },
+           |  "0"
+           |]""".stripMargin)
+  }
+
+  @Test
+  def setDestroyShouldBeIdempotent(): Unit = {
+    // Given public asset Id
+    val publicAssetId: String = createPublicAssetId()
+
+    // Destroy the public asset
+    `given`()
+      .body(
+        s"""{
+           |  "using": ["urn:ietf:params:jmap:core", "com:linagora:params:jmap:public:assets"],
+           |  "methodCalls": [
+           |    [
+           |      "PublicAsset/set", {
+           |        "accountId": "$ACCOUNT_ID",
+           |        "destroy": ["$publicAssetId"]
+           |      }, "0"
+           |    ]
+           |  ]
+           |}""".stripMargin)
+    .when()
+      .post
+    .`then`
+      .statusCode(HttpStatus.SC_OK)
+      .contentType(JSON)
+      .body("methodResponses[0][1].destroyed", hasItem(publicAssetId))
+
+    // When destroy the public asset again
+    val response: String =  `given`()
+      .body(
+        s"""{
+           |  "using": ["urn:ietf:params:jmap:core", "com:linagora:params:jmap:public:assets"],
+           |  "methodCalls": [
+           |    [
+           |      "PublicAsset/set", {
+           |        "accountId": "$ACCOUNT_ID",
+           |        "destroy": ["$publicAssetId"]
+           |      }, "0"
+           |    ]
+           |  ]
+           |}""".stripMargin)
+    .when()
+      .post
+    .`then`
+      .statusCode(HttpStatus.SC_OK)
+      .contentType(JSON)
+      .extract
+      .body()
+      .asString()
+
+    // Then the request should return destroyed
+    assertThatJson(response)
+      .inPath("methodResponses[0]")
+      .isEqualTo(
+        s"""[
+           |  "PublicAsset/set",
+           |  {
+           |    "accountId": "$ACCOUNT_ID",
+           |    "oldState": "$${json-unit.ignore}",
+           |    "newState": "$${json-unit.ignore}",
+           |    "destroyed": ["$publicAssetId"]
+           |  },
+           |  "0"
+           |]""".stripMargin)
+  }
+
+  @Test
+  def setDestroyShouldNotSupportDelegationWhenNotDelegatedUser(server: GuiceJamesServer): Unit = {
```

Comment:
Shouldn't we have a test for when it's a delegated user then? :)

Reply:
I'm not sure if this feature supports delegated users or not

Reply:
I think we can? Did we support for other methods?

Reply:
Last time I checked, we support delegation for Identity. Then it would make sense to support delegation for PublicAsset.